### PR TITLE
11893 oauth auto login

### PR DESCRIPTION
**Pull request by Nick-Triller**
See issue #11893 for the relevant proposal.

Changes:
- Implemented proposed OAuth auto login option.
- Added redirect screen because the login screen was visible for a short moment when redirecting to the OAuth endpoint.
- Updated docs.
- Fixed font awesome sizing css (e. g. fa-5x class).

### CODE REVIEW

**Comment by Nick-Triller** [comment]
The build seems to have failed because CircleCI has problems accessing npm (see [status page](https://status.circleci.com/)).

**Comment by bergquist** [comment]
Rebase onto master to get ride of the codespell errors

**Comment by Nick-Triller** [comment]
I rebased the code on master.

File: public/app/core/controllers/login_ctrl.ts
Comment (bergquist) [inline public/app/core/controllers/login_ctrl.ts]: Not sure the redirect should have handled in the frontend. Seems easier to redirect the user in the backend. Any specific reason for doing it in the frontend?

Comment (Nick-Triller) [reply public/app/core/controllers/login_ctrl.ts]: You are absolutely right, redirecting in the backend is cleaner. I will update the PR.

Comment (Nick-Triller) [reply public/app/core/controllers/login_ctrl.ts]: The PR is ready for review.

**Comment by Nick-Triller** [comment]
Do you want me to change anything else?

**Comment by bergquist** [comment]
Sorry! No maintainer had time to review this. We are quite a bit behind but we will get to this. It's a good feature, we just want to review the implementation before merging it.

**Review by bergquist — COMMENTED** [review]
Looks good! Could you please rebase this onto master? Id be happy to merge it once that's done :)

File: pkg/api/login.go
Comment (bergquist) [inline pkg/api/login.go]: This code will write twice to the ResponseWriter. first in `tryOauthAutoLogin` then in `postLoginRedirect` we redirect should only be written once.

File: pkg/api/login.go
Comment (bergquist) [inline pkg/api/login.go]: I don't think we should redirect in this case. This will cause double writes to the ResponseWriter as well

File: pkg/api/login.go
Comment (Nick-Triller) [reply pkg/api/login.go]: Thanks for the hint. The problem should be fixed now.

File: pkg/api/login.go
Comment (Nick-Triller) [reply pkg/api/login.go]: Thanks for the hint. The problem should be fixed now.

**Comment by Nick-Triller** [comment]
I rebased onto master again to include the new changes.

File: public/sass/base/font-awesome/_larger.scss
Comment (bergquist) [inline public/sass/base/font-awesome/_larger.scss]: Could you please revert this change? I'm not sure what its suppose to solve. It doesn't seem to be related to this issue.

**Review by bergquist — COMMENTED** [review]
Looks good! I will merge this once the scss change have been reverted since its not related to this feature.

File: public/sass/base/font-awesome/_larger.scss
Comment (Nick-Triller) [reply public/sass/base/font-awesome/_larger.scss]: You are right, this change is not related to the issue. I noticed it incidentally. Without the `!important`, the 3x, 4x and 5x classes don't have any effect. I will remove the commit for now.

**Review by DanCech — APPROVED** [review]
LGTM

**Comment by bergquist** [comment]
Thank you for contributing!

**Pull request merged**
